src/comment.rs

```diff
@@ -225,7 +237,13 @@ pub(crate) fn combine_strs_with_missing_comments(
         Cow::from("")
     } else {
         let one_line_width = last_line_width(prev_str) + first_line_width(&missing_comment) + 1;
-        if prefer_same_line && one_line_width <= shape.width {
+        // First comment of a comments block can be in same line if ends in the first line
+        // (and meets other conditions)
```

Comment:
can we enumerate what the other conditions are in the comment

Reply:
Changed - removed the "(and meets other conditions)" as it was redundant.  Adding the additional details could just be confusing as the "other conditions" are clear from the code.